package/gluon-core/luasrc/usr/lib/lua/gluon/wireless.lua

```diff
@@ -2,16 +2,49 @@ local sysconfig = require 'gluon.sysconfig'
 local site = require 'gluon.site'
 local util = require 'gluon.util'
 
+local unistd = require 'posix.unistd'
+local dirent = require 'posix.dirent'
+
 
 local M = {}
 
 local function find_phy_by_path(path)
-	local phy = util.glob('/sys/devices/' .. path .. '/ieee80211/phy*')[1]
-		or util.glob('/sys/devices/platform/' .. path .. '/ieee80211/phy*')[1]
+	local device_path, phy_offset
+	-- Special handling required for multi-phy devices
+	if string.match(path, '+%d$') then
+		device_path, phy_offset = string.match(path, "([^,]+)+(%d)$")
+
+		-- The UCI radio path is split into the path and a PHY offset,
+		-- which is incremented for each additional PHY available on
+		-- the device.
+	else
+		device_path = path
+		phy_offset = 0
+	end
 
-	if phy then
-		return phy:match('([^/]+)$')
+	-- Find the device path. Either it's located at /sys/devices or /sys/devices/platform
+	local path_prefix = ''
+	if not unistd.access('/sys/devices/' .. device_path .. '/ieee80211') then
+		path_prefix = 'platform/'
 	end
+
+	-- Get all available PHYs of the device and dertermine the one with the lowest index
+	local phy_names = dirent.dir('/sys/devices/' .. path_prefix .. device_path .. '/ieee80211')
+	local lowest_phy = nil
+	for _, v in ipairs(phy_names) do
+		local phy_idx = v:match('(%d)$')
+		phy_idx = tonumber(phy_idx)
+
+		if phy_idx ~= nil and (lowest_phy == nil or lowest_phy > phy_idx) then
+			lowest_phy = phy_idx
+		end
+	end
+
+	if lowest_phy == nil then
+		return nil
+	end
```

Comment:
I think it would be better to create a sorted list of all PHYs at the device and then return the entry at the correct index. This would also work when the PHY indices are not consecutive (which might happen in theory when devices appear and disappear, for example with USB WLAN adapters)

Reply:
Okay, I was misunderstanding the iwinfo code a bit. I was under the impression, the PHY index is always absolute to the lowest index, but it is not.

```
	while ((e = readdir(d)) != NULL) {
		int cur_id;

		if (strncmp(e->d_name, "phy", 3) != 0)
			continue;

		cur_id = atoi(e->d_name + 3);
		if (cur_id >= phy_id)
			continue;

		seq++;
	}
```